lib/elements/dom-if.html

```diff
@@ -235,7 +235,7 @@
       let props = this.__invalidProps;
       if (props) {
         for (let prop in props) {
-          this.__instance._setPendingProperty(prop, this.__dataHost[prop]);
+          this.__instance._setPendingProperty(prop, this.__instance.__dataHost.__dataHost[prop]);
```

Comment:
I believe this is correct, but it's awkward.

If the `dom-if` has a binding then its `dataHost` will be set and equivalent the the template `dataHost` but it not then it's unset. The template dataHost is always [set](https://github.com/Polymer/polymer/blob/master/lib/utils/templatize.html#L324) if there are any host properties.

I think we need to add an API for this `templateDataHost` or something to the `TemplateInstance` class since this is so confusing. Wdyt think @kevinpschaaf ?